test/e2e/functional/tests/e2e.go

```diff
@@ -1085,70 +939,95 @@ var _ = Describe("metallb", func() {
-				if err != nil {
-					return "", err
-				}
-				return configMap.Data[consts.MetalLBConfigMapName], err
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML(`address-pools:
+			AssertConfigMapMatchYAML(`address-pools:
 - name: addresspool1
   protocol: layer2
   addresses:
   - 1.1.1.1-1.1.1.100
-`))
+  `)
+
+			configMap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
 
 			By("Changing the Configmap")
 			configMap.Data[consts.MetalLBConfigMapName] = "fooo"
-			_, err := testclient.Client.ConfigMaps(OperatorNameSpace).Update(context.Background(), configMap, metav1.UpdateOptions{})
+			_, err = testclient.Client.ConfigMaps(OperatorNameSpace).Update(context.Background(), configMap, metav1.UpdateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Checking ConfigMap is created match the expected configuration")
-			Eventually(func() (string, error) {
-				var err error
-				configMap, err = testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				if err != nil {
-					return "", err
-				}
-				return configMap.Data[consts.MetalLBConfigMapName], err
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML(`address-pools:
+			AssertConfigMapMatchYAML(`address-pools:
 - name: addresspool1
   protocol: layer2
   addresses:
   - 1.1.1.1-1.1.1.100
-`))
+  `)
 
 			By("Deleting the Configmap")
 			err = testclient.Client.ConfigMaps(OperatorNameSpace).Delete(context.Background(), consts.MetalLBConfigMapName, metav1.DeleteOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() (string, error) {
-				var err error
-				configMap, err = testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				if err != nil {
-					return "", err
-				}
-				return configMap.Data[consts.MetalLBConfigMapName], err
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML(`address-pools:
+			AssertConfigMapMatchYAML(`address-pools:
 - name: addresspool1
   protocol: layer2
   addresses:
   - 1.1.1.1-1.1.1.100
-`))
+`)
 
 			By("Checking AddressPool resource is deleted and ConfigMap is cleared")
 			err = testclient.Client.Delete(context.Background(), addressPool)
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() string {
-				configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return configmap.Data[consts.MetalLBConfigMapName]
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML("{}"))
+			AssertConfigMapMatchYAML("{}")
 		})
 	})
 })
+
+func IsValidationWebhookEnabled() bool {
```

Comment:
No need to make this public (same for the others below)

Reply:
fixed